src/UniswapV4DeployerCompetition.sol

```diff
@@ -0,0 +1,87 @@
+// SPADIX-License-Identifier: UNLICENSED
+pragma solidity 0.8.26;
+
+import {Create2} from "@openzeppelin/contracts/utils/Create2.sol";
+import {Owned} from "solmate/src/auth/Owned.sol";
+import {VanityAddressLib} from "./libraries/VanityAddressLib.sol";
+import {IUniswapV4DeployerCompetition} from "./interfaces/IUniswapV4DeployerCompetition.sol";
+
+/// @title UniswapV4DeployerCompetition
+/// @notice A contract to crowdsource a salt for the best Uniswap V4 address
+contract UniswapV4DeployerCompetition is IUniswapV4DeployerCompetition {
+    using VanityAddressLib for address;
+
+    /// @dev The salt for the best address found so far
+    bytes32 public bestAddressSalt;
+    /// @dev The best address found so far
+    address public bestAddress;
+    /// @dev The submitter of the best address found so far
+    address public bestAddressSubmitter;
+
+    /// @dev The owner of the v4 contract
+    address public immutable v4Owner;
+    /// @dev The deadline for the competition
+    uint256 public immutable competitionDeadline;
+    /// @dev The init code hash of the V4 contract
+    bytes32 public immutable initCodeHash;
+
+    /// @dev The deployer who can initiate the deployment of the v4 PoolManager, until the exclusive deploy deadline.
+    /// @dev After this deadline anyone can deploy.
+    address public immutable deployer;
+    /// @dev The deadline for exclusive deployment by deployer after deadline
+    uint256 public immutable exclusiveDeployDeadline;
+
+    constructor(
+        bytes32 _initCodeHash,
+        address _v4Owner,
+        uint256 _competitionDeadline,
+        address _exclusiveDeployer,
+        uint256 _exclusiveDeployLength
+    ) {
+        initCodeHash = _initCodeHash;
+        v4Owner = _v4Owner;
+        competitionDeadline = _competitionDeadline;
+        exclusiveDeployDeadline = _competitionDeadline + _exclusiveDeployLength;
+        deployer = _exclusiveDeployer;
+    }
+
+    /// @inheritdoc IUniswapV4DeployerCompetition
+    function updateBestAddress(bytes32 salt) external override {
+        if (block.timestamp > competitionDeadline) {
+            revert CompetitionOver(block.timestamp, competitionDeadline);
+        }
+
+        address saltSubAddress = address(bytes20(salt));
+        if (saltSubAddress != msg.sender && saltSubAddress != address(0)) revert InvalidSender(salt, msg.sender);
+
+        address newAddress = Create2.computeAddress(salt, initCodeHash, address(this));
```

Comment:
It looks like there is a `Create2.computeAddress(salt, bytecodeHash)` function signature. It would be slightly simpler and match the `Create2.deploy` interface a little better to use that

Reply:
good catch! https://github.com/Uniswap/v4-periphery/pull/117/commits/bc2098560c23093a49ad151e716c6fc407bf1272